From cc81962349f6415e24a321fd5b4d581ba63f0ada Mon Sep 17 00:00:00 2001
From: Chenhui Zhao <chenhui.zhao@freescale.com>
Date: Wed, 22 Jan 2014 12:23:24 +0800
Subject: [PATCH 0895/1089] powerpc/suspend: disable irq by hardware when
 suspend

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

In 64-bit mode, kernel just clears the irq soft-enable flag
in struct paca_struct to disable external irqs. But, in
the case of suspend, irqs should be disabled by hardware.
Therefore, use hard_irq_disable() instead of local_irq_disable().

Change-Id: I743ec4971a9e1a9b5a3c2b463324588d48af65af
Signed-off-by: Chenhui Zhao <chenhui.zhao@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/10704
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Yang Li <LeoLi@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 arch/powerpc/kernel/time.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/kernel/time.c b/arch/powerpc/kernel/time.c
index 074bb8e..8dbe1be 100644
--- a/arch/powerpc/kernel/time.c
+++ b/arch/powerpc/kernel/time.c
@@ -576,7 +576,7 @@ static void generic_suspend_disable_irqs(void)
 	 */
 
 	set_dec(DECREMENTER_MAX);
-	local_irq_disable();
+	hard_irq_disable();
 	set_dec(DECREMENTER_MAX);
 }
 
-- 
2.0.2

